.github/workflows/rust.yml

```diff
@@ -2,21 +2,37 @@ name: Rust
 
 on:
   push:
-    branches: [ "main" ]
   pull_request:
-    branches: [ "main" ]
 
 env:
   CARGO_TERM_COLOR: always
 
 jobs:
   build:
-
     runs-on: ubuntu-latest
-
     steps:
     - uses: actions/checkout@v4
     - name: Build
       run: cargo build --verbose
     - name: Run tests
       run: cargo test --verbose
+
+  build-nix:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: cachix/install-nix-action@v31
+        with:
+          github_access_token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build
+        id: build
+        uses: mathiasvr/command-output@v2.0.0
+        with:
+          run: nix build
+      - name: Calculate cargoHash
+        if: "failure() && contains(steps.build.outputs.stderr, 'ERROR: cargoHash or cargoSha256 is out of date')"
+        run: |
+          sed -i -E 's/cargoHash = ".+";/cargoHash = "";/g' flake.nix
+          hash=$(nix build 2>&1 | sed -n -E 's/.*got: +([^ ]+).*/\1/p')
+          line=$(awk '/cargoHash/ { print NR }' flake.nix)
+          echo "::error file=flake.nix,line=$line,::cargoHash is out of date. Should be: $hash"
```

Comment:
Thanks for adding this, just to be sure it doesn't error out if cargo hash is out of date right? And just outputs stdout? might make sense to maybe add it to its own workflow then including it in rust.yml. But I can move this around later. Need to test around with rust release actions for binaries for arbitrary version tags.